### PR TITLE
Nuke testing class

### PR DESCRIPTION
**Pull request by kalisp**
Implemented pytest testing class for Nuke. 

Pulls prepared testing zip file from GDrive, prepares temporal DBs(avalon, openpype), opens Nuke, publishes basic workfile, compares result.

(in activated {OPENPYPE_ROOT}/.venv)
`{OPENPYPE_ROOT}/.venv/Scripts/python.exe {OPENPYPE_ROOT}/start.py runtests {OPENPYPE_ROOT}/tests/integration/hosts/nuke`

~(Contains merged PR for OP-2019- https://github.com/pypeclub/avalon-core/pull/391)~
~(Contains commit with current avalon-core)~

### CODE REVIEW

**Comment by mkolar** [comment]
Task linked: [OP-2019 Create test publish class for AE](https://app.clickup.com/t/6658547/OP-2019)

**Comment by mkolar** [comment]
Task linked: [OP-2042 nuke testing class](https://app.clickup.com/t/6658547/OP-2042)

File: openpype/hosts/aftereffects/plugins/publish/closeAE.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+"""Close AE after publish. For Webpublishing only."""
+import os
+
+import pyblish.api
+
+from avalon import aftereffects
```
Comment (jakubjezek001) [inline openpype/hosts/aftereffects/plugins/publish/closeAE.py:7]: shouldn't this module be part of other PR?

File: openpype/hosts/aftereffects/plugins/publish/extract_local_render.py [hunk 2]
```diff
@@ -19,10 +19,9 @@ def process(self, instance):
         staging_dir = instance.data["stagingDir"]
         self.log.info("staging_dir::{}".format(staging_dir))
```
Comment (jakubjezek001) [inline openpype/hosts/aftereffects/plugins/publish/extract_local_render.py:20]: this too is not related to Nuke, right?

File: openpype/lib/remote_publish.py [hunk 3]
```diff
@@ -11,6 +11,25 @@
 from openpype.lib.plugin_tools import parse_json
 
 
+def headless_publish(log, close_plugin_name=None, is_test=False):
```
Comment (jakubjezek001) [inline openpype/lib/remote_publish.py:14]: I believe this is not related either to Nuke or testing in general ;)

**Review by jakubjezek001 — CHANGES_REQUESTED** [review]
Lot of unrelated changes in Nuke and Testing context of this PR

**Comment by kalisp** [comment]
Cleaned up files from AE PR

**Comment by kalisp** [comment]
Comments:
- add possibility to inject application's input arguments -- will be added to test zip similar to `env_var` folder
- add class variable to inject folder with testing data - for easier building of test class, in the future to daisy chain applications
- add implicit possibility of filling `VARIANT` - if empty look for a) latest installed version b) run tests on all installed versions

**Comment by kalisp** [comment]
All issues after review implemented. 

Mentioned `-x` command line flag seems not to trigger `init.py`, still needs some testing.

**Review by jakubjezek001 — APPROVED** [review]
works

**Comment by kalisp** [comment]
Added bunch of updates:
- new wrapper class for testing in a host - each test case in a host should extend this one
- reworked db_asserts to run all asserts, created easier method to query
- output folder is set in a better way
- added `test_data_folder`, `app_variant`, `persist` to cli
- fixed test db cleanup
- fixed published folder cleanup
- test case in Nuke finally works

**Review by 64qam — APPROVED** [review]

**Comment by kalisp** [comment]
All other features will be implemented in next PRs.

**Pull request merged**